### PR TITLE
Add new `Rails/EnvironmentComparison` cop

### PR DESCRIPTION
**Pull request by tdeo**
This cop checks for comparison of `Rails.env` against strings
or symbols and replaces them with methods like `Rails.env.production?`.

**Replace this text with a summary of the changes in your PR.
The more detailed you are, the better.**

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/rails/environment_comparison.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      #   # good
+      #   Rails.env.production?
+      class EnvironmentComparison < Cop
+        MSG = 'Favor `Rails.env.%s?` over `Rails.env == %s`.'.freeze
+
+        def_node_matcher :environment_comparison?, <<-PATTERN
+          (send
+            (send (const {nil? cbase} :Rails) :env)
+            :==
+            {str sym}
```
Comment (pocke) [inline lib/rubocop/cop/rails/environment_comparison.rb]: We can use `$` to capture a node. e.g. `${str sym}` `environment_comparison?(node) do |env| ...`

File: lib/rubocop/cop/rails/environment_comparison.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      #   # bad
+      #   Rails.env == 'production'
+      #
+      #   # bad, always returns false
+      #   Rails.env == :test
+      #
+      #   # good
+      #   Rails.env.production?
+      class EnvironmentComparison < Cop
+        MSG = 'Favor `Rails.env.%s?` over `Rails.env == %s`.'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/rails/environment_comparison.rb]: I think the message should be different for string and a symbol - after all the latter is a mistake.

File: lib/rubocop/cop/rails/environment_comparison.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          (send
+            (send (const {nil? cbase} :Rails) :env)
+            :==
+            {str sym}
+          )
+        PATTERN
+
+        def on_send(node)
+          return unless environment_comparison?(node)
+          env = node.children.last
```
Comment (bbatsov) [inline lib/rubocop/cop/rails/environment_comparison.rb]: I'd replace all the usages of `children` in the code with splat destructuring of nodes. The performance is better and it's more clear what exactly are you extracting. Generally we avoid `children` everywhere in the codebase.

Comment (Drenmi) [reply lib/rubocop/cop/rails/environment_comparison.rb]: Yes, yes. I'll add an internal investigation for it. 😉

**Pull request merged**